MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm

```diff
@@ -605,18 +614,19 @@
 
 	switch (descriptorType) {
 		case VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT: {
-			bb.mtlBuffer = _mtlBuffer;
-			bb.size = _dataSize;
-			for (uint32_t i = kMVKShaderStageVertex; i < kMVKShaderStageMax; i++) {
-				if (stages[i]) {
-					bb.index = mtlIndexes.stages[i].bufferIndex + descriptorIndex;
-					if (i == kMVKShaderStageCompute) {
-						if (cmdEncoder) { cmdEncoder->_computeResourcesState.bindBuffer(bb); }
-					} else {
-						if (cmdEncoder) { cmdEncoder->_graphicsResourcesState.bindBuffer(MVKShaderStage(i), bb); }
-					}
-				}
-			}
+            bb.mtlBytes = _mtlBuffer.contents;
+            bb.size = (uint32_t)_mtlBuffer.length;
+            bb.isInline = true;
```

Comment:
If you are using `mtlBytes` and `isInline`, the effect during encoding will be to set the bytes directly into the `MTLRender/ComputeCommandEncoder`, without reference to the `MTLBuffer` in `_mtlBuffer`. Is there a reason why we would hold these bytes here in a `MTLBuffer` in `_mtlBuffer`? Could we just have `setLayout()` perform a general CPU `malloc()`?

Reply:
I was wondering about that too. Inlining the buffer is the purpose of the extension after all. So if there is no other reason to keep the `MTLBuffer` then I will replace it with a `malloc` and `free`.